predicators/approaches/maple_q_approach.py

```diff
@@ -61,14 +61,18 @@ def __init__(self, initial_predicates: Set[Predicate],
     def get_name(cls) -> str:
         return "maple_q"
 
-    def _solve(self, task: Task, timeout: int) -> Callable[[State], Action]:
+    def _solve(self,
+               task: Task,
+               timeout: int,
+               train_or_test: str = "train") -> Callable[[State], Action]:
```

Comment:
can we not have a default value here? I'm nervous about that: whoever is calling this method should explicitly specify this.

Reply:
oh it looks like we need a default in order for the type checking to work (for it to be compatible with supertype BaseApproach)